nestkernel/nestmodule.cpp

```diff
@@ -897,6 +897,26 @@ NestModule::Disconnect_g_g_D_DFunction::execute( SLIInterpreter* i ) const
   i->EStack.pop();
 }
 
+// Disconnect for arraydatum
+void
+NestModule::Disconnect_aFunction::execute( SLIInterpreter* i ) const
+{
+  i->assert_stack_load( 1 );
+
+  const ArrayDatum conns = getValue< ArrayDatum >( i->OStack.pick( 0 ) );
+
+  for ( size_t conn_index = 0; conn_index < conns.size(); ++conn_index )
+  {
+    const ConnectionDatum conn_datum = getValue< ConnectionDatum >( conns.get( conn_index ) );
+    const auto target_node = kernel().node_manager.get_node_or_proxy( conn_datum.get_target_node_id() );
+    kernel().sp_manager.disconnect(
+      conn_datum.get_source_node_id(), target_node, conn_datum.get_target_thread(), conn_datum.get_synapse_model_id() );
+  }
```

Comment:
I'd prefer if this code were in a separate function in `nest.h` and `nest.cpp`. For an example of this, see [SetStatus_idFunction](https://github.com/nest/nest-simulator/blob/d1c4393dad7f89138241a5d3981ebcb373ae7f25/nestkernel/nestmodule.cpp#L333-L344).